### ISSUE TITLE
Android app doesn't accept shared URLs unless recently opened?

### ISSUE THREAD

**Issue by mzguy** [laurent22/joplin#9850]
### Operating system

Android

### Joplin version

2.13.10

### Current behaviour

1. Something happens where Joplin enters a problematic state. For me, this happens reliably if I don't use Joplin for a while, but perhaps searching or exiting Joplin a particular way (or not exiting Joplin) does it. I'm not sure.
2. I go to Firefox and share a URL to Joplin.
3. Joplin opens showing a notebook view, but no new note is created.
4. Go back to Firefox and share the URL again. 
5. This time Joplin opens and has created a new note with the URL. This always works without fail on the 2nd attempt.
6. Further immediate sharing works, but then after waiting for some time we're back at the state in number 1 above.

This behavior started at some point with an update in the last several versions. Prior to that, sharing a URL into Joplin worked every time.

### Expected behaviour

A new note should be created with the shared URL in all cases.

### Logs

Not including the log since there is nothing useful in it. In the cases where the sharing fails, the "Received shared data" event doesn't get written to the log.

**Comment by PaulS-26** [comment]
Maybe it depends on how the app was left before?
See https://github.com/laurent22/joplin/issues/8099

**Comment by personalizedrefrigerator** [comment]
@mzguy What version of Android are you using? I'm able to reproduce #8099 in Android 7, but not Android 13.

**Comment by mzguy** [comment]
I'm on Android 14. I can't reproduce #8099 via the steps shown. I just tried it, and the first URL share failed. I exited with the back gesture, and the second URL share then worked. So, the behavior mirrors my steps, not the steps listed in #8099.

**Comment by mzguy** [comment]
@personalizedrefrigerator interestingly, I can't reproduce #9863 either. When I follow your steps, Joplin does get the shared URL, and places it in a notebook. I think the notebook it puts it into is the last one that was open prior to being put into the "All Notes" view (or, perhaps it puts it into the last notebook that received a new note or something like that). In any case, I can't reproduce that issue.

**Comment by personalizedrefrigerator** [comment]
>  When I follow your steps, Joplin does get the shared URL, and places it in a notebook.

Is Joplin being closed completely after opening "All Notes" and before sharing? By this, I mean 1) opening the running apps list and 2) swiping up/to-the-side to close the app.

**Comment by mzguy** [comment]
Yes, that's exactly what I did. Then, after sharing the note, exiting and reopening Joplin shows that the view has been switched (by the app) to the notebook the note was placed in. It switches itself out of All Notes.

**Comment by mzguy** [comment]
The issue still occurs.

**Comment by deepspaceaxolotl** [comment]
I have the same issue still.
